src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -490,19 +490,13 @@ private void OnDocEndedWithConflict(List<ReplicatedDocument> replications)
                     try {
                         Config.Database.ResolveConflict(replication.Id, Config.ConflictResolver);
                         replication = replication.ClearError();
+                    } catch (CouchbaseException e) {
+                        replication.Error = e;
                     } catch (Exception e) {
-                        C4Error error;
-                        if (e.GetBaseException() is CouchbaseException) {
-                            var errCode = ((CouchbaseException)e).Error;
-                            error = new C4Error((C4ErrorCode)errCode);
-                            replication = replication.AssignError(error, (CouchbaseException)e);
-                        } else {
-                            error = new C4Error(C4ErrorCode.UnexpectedError);
-                            replication = replication.AssignError(error,
-                                new CouchbaseLiteException(C4ErrorCode.UnexpectedError, e.Message, e));
-                        }
-                        WriteLog.To.Sync.W(Tag, $"Conflict resolution of '{replication.Id}' failed", e);
+                        replication.Error = new CouchbaseLiteException(C4ErrorCode.UnexpectedError, e.Message, e);
                     }
+                    if(replication.Error != null)
+                        WriteLog.To.Sync.W(Tag, $"Conflict resolution of '{replication.Id}' failed", replication.Error);
```

Comment:
Confirm with Pasin but I think this logging statement is not needed.

Reply:
@pasin said the chance logging this warning is very small but it's good to log it in case we encounter any edge case..
So far I see iOS has it `https://github.com/couchbase/couchbase-lite-ios/blob/master/Objective-C/CBLReplicator.mm#L599` @bmeike probably also has it... If not Blake, please log a warning message for exception if any.

Reply:
I have that same log message in my code.  It's been there forever...

Reply:
I log resolver failure, wrong db and wrong id.  The first two also get returned to the client code, as exceptions (they are returned, not thrown)